packages/waku/src/lib/utils/request.ts

```diff
@@ -22,7 +22,10 @@ export async function getInput(
   loadServerAction: (id: string) => Promise<unknown>,
 ) {
   const url = new URL(req.url);
-  url.pathname = removeBase(url.pathname, config.basePath);
+  if (config.basePath !== '/' && url.pathname.startsWith(config.basePath)) {
```

Comment:
It seems a little weird that this allows the base path to be included or excluded.

Is that how we handle it for other routes?

Reply:
> It seems a little weird that this allows the base path to be included or excluded.
> Is that how we handle it for other routes?

This is indeed odd, but that's how I did in https://github.com/wakujs/waku/pull/1699. I think I made this logic somewhat "forgiving" but ideally `url.pathname.startsWith(config.basePath)` should always hold true.

https://github.com/wakujs/waku/blob/d40ab37d8a5b71a7a06b0db0edb0e198633b2686/packages/waku/src/lib/utils/path.ts#L218-L224

Probably I was just afraid to straight away `throw new Error("pathname should start with base")` here, but it may be worth a try.